File: strings/848.ShiftingLetters.py
from typing import List


class Solution:
    def shiftingLetters(self, S: str, shifts: List[int]) -> str:
        c = 0
        s = list(S)
        l = len(shifts) - 1
        # l = 0
        for i in shifts[::-1]:
            c += i % 26
            s[l] = chr((ord(s[l]) - ord("a") + c) % 26 + ord("a"))
            l -= 1
        return "".join(s)


if __name__ == "__main__":
    # Input: S = "abc", shifts = [3,5,9]
    # Output: "rpl"
    # Explanation:
    # We start with "abc".
    s = Solution()
    print(s.shiftingLetters("abc", [3, 5, 9]))  # 'rpl'

File: strings/848.ShiftingLetters.go
package main

import (
	"fmt"
	"sort"
)

func shiftingLetters(S string, shifts []int) string {
	c := 0
	s := []rune(S)
	sort.Reverse(sort.IntSlice(shifts))
	for i := len(shifts) - 1; i >= 0; i-- {
		c += shifts[i] % 26
		s[i] = rune(((int(s[i])-int('a')+c)%26 + int('a')))
	}

	return string(s)
}

func main() {
	// Input: S = "abc", shifts = [3,5,9]
	// Output: "rpl"
	// Explanation:
	// We start with "abc".
	fmt.Println(shiftingLetters("abc", []int{3, 5, 9}))
}
